packages/lit-html/src/lit-html.ts

```diff
@@ -911,10 +911,44 @@ class ChildPartImpl {
     this._$setChildPartConnected?.(isConnected);
   }
 
+  /**
+   * The parent node into which the part renders its content.
+   *
+   * A ChildPart's content consists of a range of adjacent child nodes of
+   * `.parentNode`, possibly bordered by 'marker nodes' (`.startNode` and
+   * `.endNode`).
+   *
+   * - If both `.startNode` and `.endNode` are defined, then the part's content
+   * consists of all siblings between `.startNode` and `.endNode`, exclusively.
+   *
+   * - If `.startNode` is defined but `.endNode` is undefined, then the part's
+   * content consists of all siblings following `.startNode`, up to and
+   * including the last child of `.parentNode`. (If `.endNode` is defined, then
+   * `.startNode` will always be defined.)
+   *
+   * - If both `.endNode` and `.startNode` are undefined, then the part's
+   * content consists of all child nodes of `.parentNode`.
+   */
   get parentNode(): Node {
     return wrap(this._$startNode).parentNode!;
   }
 
+  /**
+   * The part's leading marker node, if any. See `.parentNode` for more
+   * information.
+   */
+  get startNode(): Node | undefined {
+    return this._$startNode;
+  }
+
+  /**
+   * The part's trailing marker node, if any. See `.parentNode` for more
+   * information.
+   */
+  get endNode(): Node | undefined {
+    return this._$endNode ?? undefined;
```

Comment:
Would you prefer I change the `| undefined` in the return types to `| null` also? I guess it's DOM convention and would let us avoid `?? undefined`.

Reply:
I guess I typically prefer using `undefined` as the 'true' null value thing, but this seems like a case where we could go the other way.

Reply:
It's an interesting call. In JS `undefined` is the missing value, and in the DOM `null` is. 🤬

One reason why `null` flows in here so naturally is that `null` is returned from `node.nextSibling` and a valid value for `insertBefore()`'s ref node. I'd say in this case that we don't try to force the natural type of this into our default style and type the getters as `Node | null`.

Reply:
Ok, updated, PTAL